sources/engine/Stride.Games/GameTime.cs

```diff
@@ -94,10 +97,44 @@ public GameTime(TimeSpan totalTime, TimeSpan elapsedTime)
         /// <value><c>true</c> if the <see cref="FramePerSecond"/> and <see cref="TimePerFrame"/> were updated for this frame; otherwise, <c>false</c>.</value>
         public bool FramePerSecondUpdated { get; private set; }
 
+
+
+        /// <summary>
+        /// Gets the amount of time elapsed multiplied by the time factor.
+        /// </summary>
+        /// <value>The warped elapsed time</value>
+        public TimeSpan WarpElapsed 
+        {
+            // TODO: When switching with .NET 5, use the multiply operator instead of this. elapsedGameTime * Factor
+            get => TimeSpan.FromSeconds(Elapsed.TotalSeconds * Factor);
+        }
+
+
+        /// <summary>
+        /// Gets or sets the time factor.<br/>
+        /// This value controls how much the warped time flows, this includes physics, animations and particles.
+        /// A value between 0 and 1 will slow time, a value above 1 will make it faster.
+        /// </summary>
+        /// <value>The multiply factor, a double value higher or equal to 0</value>
+        public static double Factor
```

Comment:
Todo: You could write it like so:
```cs
set => factor = value > 0 ? value : 0;
```
what do you think ?

Reply:
0 might be a bit risky because it would mean a 0-time increment per frame. this could mean that animation code that uses the delta time in as divisor will get a division by 0 error.

Reply:
I did think about that but I didn't mention it as I would expect engine code to already handle such cases: [PlayingAnimation](https://github.com/stride3d/stride/blob/master/sources/engine/Stride.Engine/Animations/PlayingAnimation.cs) already has a TimeFactor which itself can be any float value. If a system doesn't handle a delta of zero I do think it's his responsibility to work around that himself ?

Reply:
If we're thinking this way, this would make the game time factor a non-static field, but the user has to update the speed factors for each particle + animation ?

Reply:
not exactly, from what i understood you just have to make the time factor a non-static field, what tebjan and eideren are talking about is like what's happening in the 3rd person template where the animation speed is divided by the delta time so if someone makes the time factor 0 then that code will return an error for dividing by 0

Reply:
I favored Eideren's simple setter here.